### PR TITLE
Happy Thoughts API - Frida - Review wanted

### PR DESCRIPTION
**Pull request by fridamaria**
https://fridamaria-happy-api.herokuapp.com/
https://fridamaria-happy-thoughts.netlify.app/

This week was so fun! It was so great to be able to connect the frontend to the backend. I am really happy that I managed to implement a comments endpoint and connect the thought and comment models to each-other. Also happy with the frontend updates.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -2,25 +2,161 @@ import express from 'express'
...
 
-// Start defining your routes here
+// Root endpoint
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Endpoint returning 20 thoughts
+app.get('/thoughts', async (req, res) => {
```
Comment (HIPPIEKICK) [inline server.js:49]: I really like this `express-list-endpoints` but for this project the documentation says that "/" should return the 20 thoughts

File: server.js [hunk 2]
```diff
@@ -2,25 +2,161 @@ import express from 'express'
...
+
+  const sortThoughts = (sort) => {
+    if (sort === 'oldest') {
+      return { createdAt: 1 }
+    } else if (sort === 'loved') {
+      return { hearts: -1 }
+    } else {
+      return { createdAt: -1 }
+    }
+  }
```
Comment (HIPPIEKICK) [inline server.js:68]: Nice! ⭐

File: server.js [hunk 3]
```diff
@@ -2,25 +2,161 @@ import express from 'express'
...
+      })
+    }
+  } catch (err) {
+    res.status(400).json({ message: ERR_GET_THOUGHTS })
+  }
+})
+
+// Endpoint used to post new thought
+// Expects JSON body with thought message and createdBy
+app.post('/thoughts', async (req, res) => {
```
Comment (HIPPIEKICK) [inline server.js:95]: And this too should be "/"

File: server.js [hunk 4]
```diff
@@ -2,25 +2,161 @@ import express from 'express'
...
+    const thought = await new Thought({ message, createdBy }).save()
+    res.status(201).json(thought)
+  } catch (err) {
+    res.status(400).json({ message: ERR_POST_THOUGHT })
+  }
+})
+
+// Endpoint incrementing likes of given thought
+// Does not add new data to DB, therefore using POST and not PUT
+app.post('/thoughts/:id/like', async (req, res) => {
```
Comment (HIPPIEKICK) [inline server.js:108]: and this should be `/:thoughtId/like`

File: server.js [hunk 5]
```diff
@@ -2,25 +2,161 @@ import express from 'express'
...
+    const commentSent = await new Comment({ comment, createdBy, message }).save()
+
+    await Thought.findOneAndUpdate(
+      { _id: id },
+      {
+        $inc: { comment_count: 1 },
+        $push: { comments: commentSent._id }
+      },
+      { new: true }
+    )
```
Comment (HIPPIEKICK) [inline server.js:154]: Super nice extra feature! ⭐

File: server.js (hunk 1 above)
Comment (fridamaria) [reply server.js:49]: I see, but would you say that is restful?

Comment (HIPPIEKICK) [reply server.js:49]: > I really like what you've done with this project! Just take another look at the documentation and change your endpoints' names accordingly :) Apart from that - keep up the good work! ⭐

Approving both `/` and `/thoughts`, sorry for the confusion!

Comment (fridamaria) [reply server.js:49]: Yay, thank you 🎉

File: server.js (hunk 5 above)
Comment (fridamaria) [reply server.js:154]: Very happy with this, thank you 🙏

**Pull request closed without merging**